src/OpenApi/Paths.hs

```diff
@@ -30,6 +41,79 @@ newtype OperationId = OperationId Text
   deriving newtype (JSON.FromJSON, ToText)
   deriving stock   (Eq, Show)
 
+data Responses = Responses
+  { default' :: Maybe Response
+  , patterns :: Map ResponseStatusPattern Response
+  }
+  deriving stock (Eq, Show)
+
+instance JSON.FromJSON Responses where
+  parseJSON = JSON.withObject "responses" $ \object -> do
+    default' <- object .:? "default"
+    patterns <- parsePatterns . HashMap.toList $ HashMap.delete "default" object
+    pure Responses{..}
+    where
+      parsePatterns
+        :: [(Text, JSON.Value)]
+        -> JSON.Parser (Map ResponseStatusPattern Response)
+      parsePatterns pairs = Map.fromList <$> traverse (uncurry parsePair) pairs
+
+      parsePair
+        :: Text
+        -> JSON.Value
+        -> JSON.Parser (ResponseStatusPattern, Response)
+      parsePair key value =
+        (,) <$> parseResponseStatusPattern key <*> JSON.parseJSON value
+
+      parseResponseStatusPattern :: Text -> JSON.Parser ResponseStatusPattern
+      parseResponseStatusPattern input =
+        either
+          (const . fail $ "invalid status code pattern: " <> show input)
```

Comment:
I think that for consistency, the initial capitalization should match that on LOC 75 (ie. read `Invalid ...`). I'm going to fix this myself.